server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -87,6 +99,7 @@
      * Constructs a new common terms query.
      */
     public CommonTermsQueryBuilder(String fieldName, Object text) {
+        deprecationLogger.deprecated(COMMON_TERMS_QUERY_DEPRECATION_MSG);
```

Comment:
The whole class is deprecated so you don't need to log the deprecation here. Maybe add a Deprecated annotation to all constructors and you can set the deprecation for the rest handler in the SearchModule. Replacing https://github.com/elastic/elasticsearch/blob/6bba9fc83bc30d29c67e66e8e4c70f4dcc3b7494/server/src/main/java/org/elasticsearch/search/SearchModule.java#L770 with:
````
 registerQuery(new QuerySpec<>(new ParseField(CommonTermsQueryBuilder.NAME).withAllDeprecated(COMMON_TERMS_QUERY_DEPRECATION_MSG),  CommonTermsQueryBuilder::new, CommonTermsQueryBuilder::fromXContent));
````
should work.